packages/server/graphql/queries/organization.ts

```diff
@@ -12,26 +11,18 @@ export default {
       description: 'the orgId'
     }
   },
-  description: 'get a single organization and the count of users by status',
+  description: 'get a single organization',
   resolve: async (
-    {id: userId}: {id: string},
+    _source: unknown,
     {orgId}: {orgId: string},
     {authToken, dataLoader}: GQLContext
   ) => {
     const viewerId = getUserId(authToken)
-
-    const organizationUsers = await dataLoader.get('organizationUsersByUserId').load(userId)
-    const organizationUser = organizationUsers.find(
-      (organizationUser: OrganizationUser) => organizationUser.orgId === orgId
-    )
-    if (!organizationUser) return null
-    const organization = await dataLoader.get('organizations').load(orgId)
-    if (viewerId === userId) return organization
-
-    const viewerOrganizationUsers = await dataLoader.get('organizationUsersByUserId').load(userId)
-    const viewerOrganizationUser = viewerOrganizationUsers.find(
-      (organizationUser: OrganizationUser) => organizationUser.orgId === orgId
-    )
-    return viewerOrganizationUser ? organization : null
+    const [organization, viewerOrganizationUser] = await Promise.all([
+      dataLoader.get('organizations').load(orgId),
+      dataLoader.get('organizationUsersByUserIdOrgId').load({userId: viewerId, orgId})
```

Comment:
+1 While this reduces the amount of loading required here, if any of the `OrganizationUser` fields are queried from `User` we will load `organizationUsersByUserId` again. We could at least change `User.organizationUser` to use the same dataloader.
On the other hand it is unlikely to be queried this way, up to you.

Reply:
you're absolutely right. I noticed `organizationUsersByUserId` is an overfetch in most use cases since it's quickly followed by a `find`. at the risk of making this PR too big i'll go through & clean em up